### DISCUSSION TITLE
HELP- submitting my first contribution to this open source project.

### DISCUSSION THREAD

**Discussion by rramirez-dev** [mikaelacaron/Basic-Car-Maintenance#120] [General & Questions] [answered]
I have done the following:
-Fork the repo to your profile
-Clone to your computer
-Setup the upstream remote
-Created a new branch for my feature/fix i.e 106-rramirez-dv
-Made my changes and committed the changes to my branch.

What do I do next?

Do I have to set an upstream for my remote branch and then push?
    git branch --set-upstream-to=<remote>/<branch> 106-rramirez-dev

Then do I go to my branch on git hub and make a pull request?

**Comment by aaronthebald** [comment]
@rramirez-dev I would check out this video at the 55 min mark. Mikaela did a nice job of breaking it down.

**Reply by mikaelacaron** [reply]
@aaronthebald the video didn't get linked

**Reply by aaronthebald** [reply]
Whoops https://www.youtube.com/live/60ivxAOUrYU?si=RZIAn6W8shLG8A7H

I've been up since 4am im a little off my game lol

**Reply by rramirez-dev** [reply]
Got it.I was the one who ask the question about going through the process of committing a change 😂

**Comment by rramirez-dev** [comment, marked as answer]
So the final steps are as followers:
- Create a new branch
- Make your changes
- Commit your changes to your new branch
- Push your changes using one of the following methods:
   git push --set-upstream origin 106-rramirez-dev
OR
  git branch --set-upstream-to=/ 106-rramirez-dev
  git push

Navigate back to the repo on GitHub and create a pull request from your branch to the dev branch

**Reply by mikaelacaron** [reply]
you only need to set the upstream once! (and only set it to my repo, not needed for your fork, I think this is now done by default and may not be needed anymore too).

but otherwise this is correct! Thanks for posting!

**Reply by rramirez-dev** [reply]
When I tried to push my changes it told me I needed to use git push --set-upstream origin 106-rramirez-dev
or set the upstream to my branch.

**Reply by mikaelacaron** [reply]
What's `git remote -v` say? (this lists all the remotes)

**Reply by rramirez-dev** [reply]
I deleted my branch before I ran this.

origin	https://github.com/rramirez-dev/Basic-Car-Maintenance.git (fetch)
origin	https://github.com/rramirez-dev/Basic-Car-Maintenance.git (push)
upstream	https://github.com/mikaelacaron/Basic-Car-Maintenance.git (fetch)
upstream	https://github.com/mikaelacaron/Basic-Car-Maintenance.git (push)

**Reply by mikaelacaron** [reply]
> When I tried to push my changes it told me I needed to use git push --set-upstream origin 106-rramirez-dev
or set the upstream to my branch.

This would have happened if all you did was type `git push`

This would *not* have happened if you did `git push origin 106-rramirez-dev`

**Reply by rramirez-dev** [reply]
So after I commit my changes to my branch I should do "git push origin 106-rramirez-dev" and not just "git push"?

**Reply by mikaelacaron** [reply]
So how it works is you only need to "set the upstream" once and after you've done it, you don't need to do it again for that branch.
But if you make a new branch, you need to set the upstream again for that branch. And actually looking at this now "upstream" refers to two different things, depending on the context.


think of "setting the upstream" meaning tracking something else. So the two scenarios are:
* "setting the upstream" to track my repo (the original repo you created your fork from)
   * Which technically you actually don't need to do, but it just depends how you sync your changes, there's always multiple ways to do things 🫨
* "setting the upstream" to track your own remote repo meaning `origin 106-rramirez-dev`
   * `origin` is the name of the remote, and you have to push the branch `106-rramirez-dev` to GitHub


Not sure if this helps, haha or confuses you more, but I'd suggest looking up Git / GitHub tutorials and resources related to open source
